pkgs/by-name/co/cook-cli/package.nix

```
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, buildNpmPackage
+, rustPlatform
+, pkg-config
+, openssl
+, darwin
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "cook-cli";
+  version = "0.7.1";
+
+  src = fetchFromGitHub {
+    owner = "cooklang";
+    repo = "cookcli";
+    rev = "v${version}";
+    hash = "sha256-3gLVsk6GCxOG24Md7E9fk28Vnc4kVDdwyZUD/GtSwFE=";
+  };
+
+  cargoHash = "sha256-6lnURuE1cgNAniHl5ozXo1W3cLYYje7er+ZhvZDKdVg=";
+
+  nativeBuildInputs = [ pkg-config openssl ];
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
+
+  postPatch = ''
+    cp -r ${passthru.ui}/* "ui/public"
```

```suggestion
    ln -s ${passthru.ui} "ui/public"
```
because you will have the ui derivation in the nix store anyways and if you copy the files you have them twice, once in the ui derivation and a second time vendroed in the cook-cli derivation.

---

Yeah this makes sense. I had to do `rm -rf "ui/public"` before making the symbolic link.